emily_cron/app/models/deposit.py

```diff
@@ -0,0 +1,163 @@
+from dataclasses import asdict, dataclass
+from datetime import datetime
+from enum import Enum
+import functools
+from typing import Any, Optional, Self
+
+from bitcoinlib.scripts import Script
+
+from app import settings
+
+
+class RequestStatus(Enum):
+    PENDING = "pending"
+    ACCEPTED = "accepted"
+    CONFIRMED = "confirmed"
+    FAILED = "failed"
+    REPROCESSING = "reprocessing"
+
+
+@dataclass
+class DepositInfo:
+    """Represents a deposit transaction."""
+
+    bitcoin_txid: str
+    bitcoin_tx_output_index: int
+    recipient: str
+    amount: int
+    last_update_height: int
+    last_update_block_hash: str
+    status: str
+    reclaim_script: str
+    deposit_script: str
+
+    @classmethod
+    def from_json(cls, j: dict[str, Any]) -> Self:
+        return cls(
+            bitcoin_txid=j["bitcoinTxid"],
+            bitcoin_tx_output_index=j["bitcoinTxOutputIndex"],
+            recipient=j["recipient"],
+            amount=j["amount"],
+            last_update_height=j["lastUpdateHeight"],
+            last_update_block_hash=j["lastUpdateBlockHash"],
+            status=j["status"],
+            reclaim_script=j["reclaimScript"],
+            deposit_script=j["depositScript"],
+        )
+
+    @property
+    def lock_time(self) -> int:
+        """Extracts lock time from reclaim script."""
+        script = Script.parse(self.reclaim_script)
+        op_code_maybe = script.view().split()[0]
+        if op_code_maybe.startswith("OP_"):
+            return int(op_code_maybe[len("OP_") :])
+        return int.from_bytes(script.commands[0], byteorder="little")
+
+    @property
+    def max_fee(self) -> int:
+        """Extracts the max fee from deposit script."""
+        script = Script.parse(self.deposit_script)
+        if script.redeemscript:
+            max_fee_bytes = script.redeemscript
+        else:
+            max_fee_bytes = bytes.fromhex(script.view().split()[0])
+        return int.from_bytes(max_fee_bytes[:8], byteorder="big")
```

Comment:
Are both of these `int.from_bytes` correct? Python integers are signed, but the `max_fee` and the `locktime` integers are both signed. For `locktime`s I suspect that with our current code it doesn't matter in our case, since python represents integers using the two's complement layout. With the expected layout of the `locktime` this shouldn't be an issue. But pointing it out to double check.

Reply:
From the docs (https://docs.python.org/3/library/stdtypes.html#int.from_bytes), by default `signed=False`, so this is producing only unsigned values -- which could be wrong for the lock time IIUC. 
While I guess the second one is correct since that comes from "our" encoding, and we want it unsigned.

Reply:
mmm apparently bitcoin uses a different encoding. Looking at https://bitcoin.stackexchange.com/questions/122939/understanding-scripts-number-encoding-cscriptnum, taking the first example (`[0x32; 0x79; 0x86]`), the bitcoin code does produce the claimed value (https://onecompiler.com/cpp/43e89drzc, taken from https://github.com/bitcoin/bitcoin/blob/cfe025ff0e1dcf984179a5a9b01dc8b865d95409/src/script/script.h#L383), while python produces something different:
```
>>> int.from_bytes(bytes([0x32, 0x79, 0x86]), byteorder="little", signed=True)
-7964366
```

Reply:
And shouldn't the max fee also be `signed=False`?

Reply:
Nice! Can you also add a comment with the source of that code in `decode_cscript_int`?